chef/cookbooks/nova/recipes/compute.rb

```diff
@@ -423,3 +429,36 @@ def set_boot_kernel_and_trigger_reboot(flavor='default')
 execute "trigger-nova-own-az-config" do
   command "true"
 end
+
+# Set iptables rules for blocking VNC Access for all but the nova-controller node.
+bash "nova_compute_vncblock_reject_all" do
+  code <<-EOH
+    iptables -D INPUT -j VNCBLOCK -p tcp --match multiport --dports 5900:15900
+    iptables -I INPUT -j VNCBLOCK -p tcp --match multiport --dports 5900:15900
+    iptables -F VNCBLOCK
+    iptables -I VNCBLOCK -p tcp \
+        -m connbytes --connbytes 0:1024 \
+        --connbytes-dir both --connbytes-mode bytes \
+        -m state --state ESTABLISHED \
+        -m u32 --u32 "0>>22&0x3C@ 12>>26&0x3C@ 0=0x52464220" \
```

Comment:
@bmwiedemann I did it on purpose to check for VNC match only for those packets which are under the port range.